### PR TITLE
Add response codes for additional conditions

### LINKED ISSUE #1700: Finalize response codes for the various invalid specifications of a custom fee
**Summary**
Known edge cases without a well-defined response code include,
- [x] A fractional fee with a specified maximum smaller than its specified minimum.
- [x] A fractional fee whose denominator and numerator are both negative (maybe acceptable, maybe `CUSTOM_FEES_MUST_BE_POSITIVE`?).

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
**Related issue(s)**:
Closes #1700 

**Summary of the change**:
1. When a fractional fee is specified with maximum_amount smaller than minimum_amount, it results in `FRACTIONAL_FEE_MAX_AMOUNT_LESS_THAN_MIN_AMOUNT`.
2.  A fractional fee whose denominator and numerator are both negative , results in `CUSTOM_FEES_MUST_BE_POSITIVE`.

### CODE REVIEW

File: hedera-node/src/test/java/com/hedera/services/store/tokens/HederaTokenStoreTest.java [hunk 1]
```diff
@@ -262,7 +263,33 @@
...
+					)).build();
+	private CustomFeesOuterClass.CustomFees grpcMaxEqualToMinCustomFees = CustomFeesOuterClass.CustomFees.newBuilder()
+			.addCustomFees(CustomFeesOuterClass.CustomFee.newBuilder()
+					.setFeeCollectorAccountId(feeCollector)
+					.setFractionalFee(CustomFeesOuterClass.FractionalFee.newBuilder()
+							.setFractionalAmount(Fraction.newBuilder()
+									.setNumerator(123L).setDenominator(1_000L))
+							.setMaximumAmount(2L)
+							.setMinimumAmount(2L)
+							.setMinimumAmount(2L)
```
Comment (abhishek-hedera) [inline hedera-node/src/test/java/com/hedera/services/store/tokens/HederaTokenStoreTest.java:283]: `setMinimumAmount(2L)` is repeated twice here

Comment (ljianghedera) [reply hedera-node/src/test/java/com/hedera/services/store/tokens/HederaTokenStoreTest.java:283]: Though it won't hurt, but may cause some confusion.

**Review by ljianghedera — DISMISSED** [review]
LGTM overall.

File: hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java [hunk 2]
```diff
@@ -466,8 +470,8 @@ private ResponseCodeEnum validateFeeSchedule(List<CustomFeesOuterClass.CustomFee
 		return OK;
 	}
 
-	private boolean signsMatch(long a, long b) {
-		return (a < 0 && b < 0) || (a > 0 && b > 0);
+	private boolean areValidPositiveNumbers(long a, long b) {
+		return (a > 0 && b > 0);
```
Comment (qnswirlds) [inline hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java:474]: So we don't allow (-25/-100) any more? Just (25/100)?

**Review by qnswirlds — APPROVED** [review]
I pushed a few more changes as my comments.

**Comment by Neeharika-Sompalli** [comment]
> I pushed a few more changes as my comments.

@qnswirlds  Thank you for the changes !

File: hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java (hunk 2 above)
Comment (Neeharika-Sompalli) [reply hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java:538]: @qnswirlds  yes we allow only positive numbers

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenUpdateSpecs.java [hunk 3]
```diff
@@ -677,6 +679,28 @@ private HapiApiSpec onlyValidCustomFeeScheduleCanBeUpdated() {
...
+										numerator, denominator,
+										minimumToCollect * 20, OptionalLong.of(maximumToCollect),
+										tokenCollector))
+								.hasKnownStatus(FRACTIONAL_FEE_MAX_AMOUNT_LESS_THAN_MIN_AMOUNT),
+						tokenUpdate(token)
+								.treasury(tokenCollector)
+								.customFeesMutable(true)
+								.withCustom(fractionalFee(
+										numerator, denominator,
+										minimumToCollect * 10, OptionalLong.of(maximumToCollect),
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenUpdateSpecs.java]: @ljianghedera  Thank you!  @qnswirlds  had fixed it to be clearer. The multiplication was done to make minimum and maximum amounts equal 😃

File: hedera-node/src/test/java/com/hedera/services/store/tokens/HederaTokenStoreTest.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/src/test/java/com/hedera/services/store/tokens/HederaTokenStoreTest.java]: @abhishek-hedera  @ljianghedera  Thank you ! It is removed

**Pull request merged**